packages/spacecat-shared-data-access/src/models/api-key/api-key.js

```diff
@@ -43,6 +43,28 @@ const ApiKey = (data) => {
   self.getDeletedAt = () => self.state.deletedAt;
   self.getScopes = () => self.state.scopes;
 
+  /**
+   * Checks if the apiKey is valid.
+   * @returns {boolean} True if the apiKey is valid, false otherwise
+   */
+  self.isValid = () => {
+    const now = new Date();
+
+    if (self.state.deletedAt && new Date(self.state.deletedAt) > now) {
```

Comment:
Is this comparison correct? If deletedAt is > now, that would mean it was deleted in the future — right? And therefore still valid?

Reply:
Yes. Correct.

Reply:
It will be deleted in future. So, the key is still valid.